plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1024,9 +1024,19 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
     private Optional<Long> getSnapshotId(org.apache.iceberg.Table table, Optional<Long> snapshotId)
     {
         // table.name() is an encoded version of SchemaTableName
-        return snapshotIds.computeIfAbsent(table.name(), ignored -> snapshotId
-                .map(id -> IcebergUtil.resolveSnapshotId(table, id))
-                .or(() -> Optional.ofNullable(table.currentSnapshot()).map(Snapshot::snapshotId)));
+        return snapshotId.or(() -> Optional.ofNullable(table.currentSnapshot()).map(Snapshot::snapshotId))
+                .map(id ->
+                        snapshotIds.computeIfAbsent(
+                                createId(table.name(), id),
+                                ignored -> IcebergUtil.resolveSnapshotId(table, id)));
```

Comment:
It feels like we will call `resolveSnapshodId` for `table.currentSnapshot` with new code. Is that a problem? Can we avoid that?